pandas/core/sparse/scipy_sparse.py

```diff
@@ -129,23 +129,29 @@ def _coo_to_sparse_series(A, dense_index: bool = False,
 
     Returns
     -------
-    Series or SparseSeries
+    Series or SparseSeries on success
+    TypeError if A is not a coo_matrix
+
     """
     from pandas import SparseDtype
 
-    s = Series(A.data, MultiIndex.from_arrays((A.row, A.col)))
-    s = s.sort_index()
-    if sparse_series:
-        # TODO(SparseSeries): remove this and the sparse_series keyword.
-        # This is just here to avoid a DeprecationWarning when
-        # _coo_to_sparse_series is called via Series.sparse.from_coo
-        s = s.to_sparse()  # TODO: specify kind?
-    else:
-        s = s.astype(SparseDtype(s.dtype))
-    if dense_index:
-        # is there a better constructor method to use here?
-        i = range(A.shape[0])
-        j = range(A.shape[1])
-        ind = MultiIndex.from_product([i, j])
-        s = s.reindex(ind)
-    return s
+    try:
+        s = Series(A.data, MultiIndex.from_arrays((A.row, A.col)))
+        s = s.sort_index()
+        if sparse_series:
+            # TODO(SparseSeries): remove this and the sparse_series keyword.
+            # This is just here to avoid a DeprecationWarning when
+            # _coo_to_sparse_series is called via Series.sparse.from_coo
+            s = s.to_sparse()  # TODO: specify kind?
+        else:
+            s = s.astype(SparseDtype(s.dtype))
+        if dense_index:
+            # is there a better constructor method to use here?
+            i = range(A.shape[0])
+            j = range(A.shape[1])
+            ind = MultiIndex.from_product([i, j])
+            s = s.reindex(ind)
+        return s
```

Comment:
son't put the try/except around everything, narrow this down to the first possible place
(which is likely
``` 
try:
    s= Series(A.data, MultiIndex.from_arrays((A.row, A.col)))
except AttributeError:
    .....
```